crates/matrix-sdk-ui/src/timeline/event_handler.rs

```diff
@@ -304,7 +305,15 @@ impl<'a, 'o> TimelineEventHandler<'a, 'o> {
                 ) => self.handle_poll_start_edit(c.relates_to),
                 AnyMessageLikeEventContent::UnstablePollStart(
                     UnstablePollStartEventContent::New(c),
-                ) => self.handle_poll_start(c, should_add),
+                ) => {
+                    let mut poll_state = PollState::new(c);
+                    if let Flow::Remote { event_id, .. } = self.ctx.flow.clone() {
+                        // Applying the cache to remote events only because local echoes
+                        // don't have an event ID that could be referenced by responses yet.
+                        self.meta.poll_pending_events.apply(&event_id, &mut poll_state);
+                    }
+                    self.add(should_add, TimelineItemContent::Poll(poll_state));
+                }
```

Comment:
This makes a long function longer, and further embeds the fact that it has 2 jobs: finding what `event_kind` is, and dealing with each type it might be.

If you are looking for more regularity, I would prefer making _every_ case call a function, instead of _none_ ...

Reply:
Alright, I was a bit torn about this one myself, so I'm happy to partially revert it: I'll keep the inlining of handling an encrypted event, since I don't see the value of a one-liner as a separate function called only once.

Thanks for the review!

Reply:
Well, it's a one-liner *now*, but it holds useful tracing context and there's a todo comment there implying that more complex logic will be added in the future.

Reply:
Either way is fine with me.

Reply:
This one-liner has been a one-liner for almost a year (last touched in Feb 2023), and it's easy to revert that when the time comes, so I'll keep it like this.